superset/dashboards/dao.py

```diff
@@ -35,6 +36,14 @@ class DashboardDAO(BaseDAO):
     model_cls = Dashboard
     base_filter = DashboardFilter
 
+    @staticmethod
+    def get_charts_for_dashboard(dashboard_id: int) -> List[Slice]:
+        query = db.session.query(Dashboard).filter(Dashboard.id == dashboard_id)
+        dashboard = query.scalar()
+        if not dashboard:
+            raise DashboardNotFoundError()
+        return [slice.data for slice in dashboard.slices]
```

Comment:
This will trigger N+1 queries on the DB. A possible solution can be `dashboard = db.session.query(Dashboard).join(Slice, Dashboard.slices).join(Slice.table).filter(Dashboard.id == 1).options(contains_eager(Dashboard.slices).contains_eager(Slice.owners)).one_or_none()`

I would avoid using `slice.data` unless it's really handy on this case

Reply:
Interesting. Is it calling `dashboard.slices` that triggers many queries, or `slice.data` in a loop?

I called `slice.data` to get it into a JSON serializable format. Is that not the accepted way to do that with a SQLAlchemy model?

Reply:
The extra queries were on both, while iterating the relation and when `Slice.data` accesses owners. SQLAlchemy lazy loads by default.

I you mean generically no, `slice.data` is a specific custom Slice model method/property. If you mean the accepted way of getting a `Dict` from a slice model on superset yes, but since we are laying the ground here for a future SPA, I prefer using marshmallow to serialise SQLA objects.

Do do you think?

Reply:
If we defaults to marshmallow for object serialization, does that mean `x.data` will be eventually deprecated someday?

Reply:
For now it's just a personal preference. I don't want to block this PR. Probably for now it's better to just solve the N+1 problem and keep the using slice.data.

Reply:
I've addressed this by using something similar to the suggested query, as well as adding a Marshmallow schema that is somewhat different from the existing `bootstrap_data` schema to remove a few less-REST-ful fields.